dweather_client/client.py

```diff
@@ -648,9 +648,30 @@ def get_afr_history(ipfs_timeout=None):
 def has_dataset_updated(dataset, slices, as_of, ipfs_timeout=None):
     """
     Determine whether any dataset updates generated after `as_of` affect any `slices` of date ranges.
-    """
+    """    
     with DateRangeRetriever(dataset, ipfs_timeout=ipfs_timeout) as dataset_obj:
         ranges = dataset_obj.get_data(as_of)
     return has_changed(slices, ranges)
 
-
+def get_teleconnections_history(weather_variable, ipfs_timeout=None):
+    with TeleconnectionsDataset(ipfs_timeout=ipfs_timeout) as dataset_obj:
+        csv_text = dataset_obj.get_data()
+        history = {}
+        reader = csv.reader(csv_text.split('\n'))
+        headers = next(reader)
+        date_col = headers.index('DATE')
+        try:
```

Comment:
Comments explaining the steps in this would be helpful for maintainers or copiers. I realize that hasn’t been the standard until now but I was unable to decipher this cleanly without messaging you on slack.

Reply:
That goes for all the try/excepts below